### PR TITLE
Issue/76 lookup resolve domain addressinput

### PR DESCRIPTION
**Pull request by haolinj**
#76

### CODE REVIEW

File: code/client/src/components/AddressInput.jsx [hunk 1]
```diff
@@ -24,13 +34,53 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
...
+            domainName,
+            filterValue: searchValue,
+            selected: false
+          })
+        }
+
+        if (!isEmpty(searchValue)) {
+          const resolvedAddress = await api.blockchain.domain.resolve({ name: searchValue })
+
+          if (resolvedAddress && resolvedAddress !== '0x0000000000000000000000000000000000000000') {
```
Comment (polymorpher) [inline code/client/src/components/AddressInput.jsx:65]: util.isEmptyAddress

File: code/client/src/components/AddressInput.jsx [hunk 2]
```diff
@@ -24,13 +34,53 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
...
+
+          setAddressCallback({
+            value: validAddress,
+            domainName,
+            filterValue: searchValue,
+            selected: false
+          })
+        }
+
+        if (!isEmpty(searchValue)) {
```
Comment (polymorpher) [inline code/client/src/components/AddressInput.jsx:62]: could do some preliminary filtering to reduce queries to blockchain, e.g. `searchValue.includes(`${ONEConstants.Domain.DEFAULT_PARENT_LABEL}.${ONEConstants.Domain.DEFAULT_TLD} )`

File: code/client/src/components/AddressInput.jsx [hunk 3]
```diff
@@ -24,13 +34,53 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
...
+              value: resolvedAddress,
+              domainName: searchValue,
+              filterValue: searchValue,
+              selected: false
+            })
+          }
+        }
+
+        setSearchingAddress(false)
+      } catch (e) {
```
Comment (polymorpher) [inline code/client/src/components/AddressInput.jsx:76]: let's log it to console (which should be also captured by Sentry) so we know if the resolver is temporarily down or if we are blocked due to 429

File: code/client/src/components/AddressInput.jsx [hunk 4]
```diff
@@ -42,27 +92,48 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
...
-    // Init the known address entries for existing wallets.
-    walletsNotInKnownAddresses.forEach((wallet) => {
-      dispatch(walletActions.setKnownAddress({
-        label: wallet.name,
-        address: wallet.address,
-        network: wallet.network,
-        creationTime: wallet.effectiveTime,
-        numUsed: 0
+      const knownAddressesWithoutDomain = existingKnownAddresses.filter((knownAddress) =>
+        !isEmpty(existingKnownAddresses.domain?.name)
```
Comment (polymorpher) [inline code/client/src/components/AddressInput.jsx:105]: typo?

Comment (haolinj) [reply code/client/src/components/AddressInput.jsx]: Yes, fixed.

File: code/client/src/components/AddressInput.jsx [hunk 5]
```diff
@@ -42,27 +94,48 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
...
 
-    // Init the known address entries for existing wallets.
-    walletsNotInKnownAddresses.forEach((wallet) => {
-      dispatch(walletActions.setKnownAddress({
-        label: wallet.name,
-        address: wallet.address,
-        network: wallet.network,
-        creationTime: wallet.effectiveTime,
-        numUsed: 0
+      const knownAddressesWithoutDomain = existingKnownAddresses.filter((knownAddress) =>
```
Comment (polymorpher) [inline code/client/src/components/AddressInput.jsx:106]: we should probably add a time condition later to reduce queries, but it is probably fine for now

Comment (haolinj) [reply code/client/src/components/AddressInput.jsx:106]: Yeah, I agree, right now this is just bootstrap existing address that does not have a domain. We can use lookupTime or last lookupTime to limit the query.

**Comment by polymorpher** [comment]
Looks good. Doing some tests.

**Comment by polymorpher** [comment]
`Enter` should be as good as select. Right now it clears selection

**Comment by polymorpher** [comment]
If I type the domain name of a known wallet (e.g. another 1wallet of mine), it fails to resolve the domain correctly to that wallet.

image
 
image

**Comment by polymorpher** [comment]
It should also automatically trim extra spaces

image

**Comment by polymorpher** [comment]
Should add support to Restore as well

image

**Comment by polymorpher** [comment]
The rest seems to work very well

**Comment by haolinj** [comment]
> If I type the domain name of a known wallet (e.g. another 1wallet of mine), it fails to resolve the domain correctly to that wallet.
> 
> image
> 
> image

This is due to no trim to the input address value, should be fixed.

**Comment by polymorpher** [comment]
Restore: Showing technical errors on mainnet if address is not contract - the error should be hidden (should not use the handler in safeExec); Not recognizing address on Ganache (because of exception in api call)

image


image

**Comment by polymorpher** [comment]
Looks good

**Pull request merged**